src/commands.rs

```diff
@@ -3982,6 +3993,56 @@ fn cmd_workspace_list(
     Ok(())
 }
 
+fn cmd_workspace_update_stale(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    _args: &WorkspaceUpdateStaleArgs,
+) -> Result<(), CommandError> {
+    let workspace = command.load_workspace(ui)?;
+    let mut workspace_command = command.resolve_operation(ui, workspace)?;
+    workspace_command.check_working_copy_writable()?;
+    let repo = workspace_command.repo().clone();
+    let workspace_id = workspace_command.workspace_id();
+    let desired_wc_commit_id = match repo.view().get_wc_commit_id(&workspace_id) {
+        Some(wc_commit_id) => wc_commit_id.clone(),
+        None => {
+            return Err(user_error("This command requires a working copy"));
+        }
+    };
+    let (mut locked_wc, _wc_commit) = workspace_command.unsafe_start_working_copy_mutation()?;
+    let desired_wc_commit = repo.store().get_commit(&desired_wc_commit_id)?;
```

Comment:
Perhaps, `workspace_command.check_working_copy_writable()` and `desired_wc_commit_id` lookup are redundant. Maybe we can reuse result of `unsafe_start_working_copy_mutation()`, or make it more primitive function that just calls `start_mutation()`.

Reply:
Oh, good catch! I did the former.